api4/channel.go

```diff
@@ -385,6 +386,26 @@ func getPublicChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Request
 	}
 }
 
+func getDeletedChannelsForTeam(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	if !app.SessionHasPermissionToTeam(c.Session, c.Params.TeamId, model.PERMISSION_MANAGE_TEAM) {
+		c.SetPermissionError(model.PERMISSION_MANAGE_TEAM)
+		return
+	}
+
+	if channels, err := app.GetDeletedChannels(c.Params.TeamId, c.Params.Page, c.Params.PerPage); err != nil {
```

Comment:
I think the second argument should be `c.Params.Page * c.Params.PerPage` in order to convert from page/per-page to offset/limit.